types/react-query/index.d.ts

```diff
@@ -36,53 +47,62 @@ export interface QueryOptions<TResult> {
-    paginated: true;
-    getCanFetchMore: (lastPage: TResult, allPages: TResult[]) => boolean;
+export interface InfiniteQueryOptions<TResult, TVariables> extends QueryOptions<TResult> {
+    getFetchMore?: ((lastGroup: TResult, allGroups: TResult[]) => OmitKey<TVariables>) | boolean
 }
 
 export interface QueryResult<TResult, TVariables> {
-    data: null | TResult;
+    status: 'loading' | 'error' | 'success'
+    data: undefined | TResult[];
     error: null | Error;
-    isLoading: boolean;
     isFetching: boolean;
-    isCached: boolean;
     failureCount: number;
-    refetch: (arg?: {variables?: TVariables, merge?: (...args: any[]) => any, disableThrow?: boolean}) => Promise<void>;
+    refetch: (arg?: {variables?: OmitKey<TVariables>, merge?: (...args: any[]) => any, disableThrow?: boolean}) => Promise<any>;
 }
 
-export interface QueryResultPaginated<TResult, TVariables> extends QueryResult<TResult[], TVariables> {
-    isFetchingMore: boolean;
-    canFetchMore: boolean;
-    fetchMore: (variables?: TVariables) => Promise<TResult>;
+export interface QueryResultPaginated<TResult, TVariables> {
+    status: 'loading' | 'error' | 'success'
+    resolvedData: undefined | TResult;
+    latestData: undefined | TResult;
+    error: null | Error;
+    isFetching: boolean;
+    failureCount: number;
+    refetch: (arg?: {variables?: OmitKey<TVariables>, merge?: (...args: any[]) => any, disableThrow?: boolean}) => Promise<any>;
 }
 
-export function prefetchQuery<TResult, TVariables extends object>(
-    queryKey: QueryKey<TVariables>,
-    queryFn: QueryFunction<TResult, TVariables>,
-    options?: PrefetchQueryOptions<TResult>
-): Promise<TResult>;
-
-export interface PrefetchQueryOptions<TResult> extends QueryOptions<TResult> {
-    force?: boolean;
+export interface QueryResultInfinite<TResult, TVariables> {
+    status: 'loading' | 'error' | 'success'
+    data: TResult[];
+    error: null | Error;
+    isFetching: boolean;
+    isFetchingMore: boolean;
+    failureCount: number;
+    refetch: (arg?: {variables?: OmitKey<TVariables>, merge?: (...args: any[]) => any, disableThrow?: boolean}) => Promise<any>;
+    fetchMore: (variables?: OmitKey<TVariables>) => Promise<any>;
+    canFetchMore: boolean
 }
 
 export function useMutation<TResults, TVariables extends object>(
     mutationFn: MutationFunction<TResults, TVariables>,
-    mutationOptions?: MutationOptions
+    mutationOptions?: MutationOptions<TResults>
 ): [MutateFunction<TResults, TVariables>, MutationResult<TResults>];
 
 export type MutationFunction<TResults, TVariables extends object> = (
     variables: TVariables,
 ) => Promise<TResults>;
 
-export interface MutationOptions {
-    refetchQueries?: Array<string | [string, object]>;
-    refetchQueriesOnFailure?: boolean;
+export interface MutationOptions<TResult> {
+    onSuccess?: (data: TResult) => (Promise<any> | undefined),
+    onSettled?: (data: TResult, error: any) => (Promise<any> | undefined),
```

Comment:
why not `| void`

Reply:
https://github.com/Microsoft/dtslint/blob/master/docs/void-return.md